### ISSUE TITLE
Remove exception value in `try..catch`, if it's not used

### ISSUE THREAD

**Issue by shoonia** [terser/terser#1507]
**Feature request**

Remove exception value in `try..catch`, if it's not used.

```diff
function isValidJSON(text) {
  try {
    JSON.parse(text);
    return true;
-  } catch (error) {
+  } catch {
    return false;
  }
}
```

**Comment by ehoogeveen-medweb** [comment]
Good idea :) Optional catch binding is an ES2019 feature, so would need to be gated behind `ecma: 2019` or higher:
https://compat-table.github.io/compat-table/es2016plus/
https://caniuse.com/mdn-javascript_statements_try_catch_optional_catch_binding

**Comment by fabiosantoscode** [comment]
The `ecma` option is poorly understood, so I believe it could `catch` some people by surprise. I think a new option is needed, defaulted to false.
